server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -395,7 +395,14 @@ public void onNewAcquiredCommit(final IndexCommit commit, final Set<String> addi
                     final IndexCommitRef indexCommitRef = acquireIndexCommitRef(() -> commit);
                     var primaryTerm = config().getPrimaryTermSupplier().getAsLong();
                     assert indexCommitRef.getIndexCommit() == commit;
-                    wrappedListener.onNewCommit(shardId, store, primaryTerm, indexCommitRef, additionalFiles);
+                    wrappedListener.onNewCommit(
+                        shardId,
+                        store,
+                        primaryTerm,
+                        indexCommitRef,
+                        additionalFiles,
+                        getCommitExtraTransientData()
```

Comment:
Does it make sense to wrap it with `Collections.unmodifiableMap` or do we expect that `getCommitExtraTransientData` never returns a modifiable map?